src/beamlime/applications/raw_count_handlers.py

```diff
@@ -91,20 +101,30 @@ def handle(self, message: DataPieceReceived) -> WorkflowResultUpdate | None:
             for name in det:
                 buffer = self._buffer[name]
                 buffer.append(event_id)
-                self._pulse += 1
         else:
             self.info("Ignoring data for %s", detname)
-            return
-        npulse = 10
-        if self._pulse % npulse == 0:
+        reference_time = sc.datetime(int(data_piece.reference_time), unit='ns')
+        if self._next_update is None:
+            self._next_update = reference_time.copy()
+        if reference_time >= self._next_update:
+            self.info("Updating views at %s", reference_time)
+            while reference_time >= self._next_update:
+                # If there were no pulses for a while we need to skip several updates.
+                # Note that we do not simply set _next_update based on reference_time
+                # to avoid drifts.
+                self._next_update += self._update_every
```

Comment:
```suggestion
            self._next_update += ((reference_time - self._next_update) // self._update_every + 1) * self._update_every
```
this should avoid the loop?